referenceProvider.js

```diff
@@ -0,0 +1,128 @@
+const vscode = require('vscode');
+const parser = require('./parser');
```

Comment:
On the second thought, I didnt like this new module scheme. The original file is already small. Could you only provide fixes on the main file? I will then think of a module structure.